contracts/staking/Staking.sol

```diff
@@ -1004,54 +978,67 @@ contract Staking is StakingV2Storage, GraphUpgradeable, IStaking, Multicall {
             );
             queryFees = queryFees.sub(curationFees);
 
-            // Add funds to the allocation
+            // -- Process rebate reward --
+            // Using accumulated fees and subtracting previously distributed rebates
+            // allows for multiple vouchers to be collected while following the rebate formula
             alloc.collectedFees = alloc.collectedFees.add(queryFees);
+            uint256 newRebates = LibExponential.exponentialRebates(
+                alloc.collectedFees,
+                alloc.tokens,
+                alphaNumerator,
+                alphaDenominator,
+                lambdaNumerator,
+                lambdaDenominator
+            );
+
+            //  -- Ensure rebates to distribute are within bounds --
+            // Indexers can become under or over rebated if rebate parameters (alpha, lambda)
```

Comment:
It's not possible for it to happen if parameters remain constant, using the accumulated query fees in the formula instead of the query fees being redeemed ensures that.